### PR TITLE
Remove `targetReferences` from any successful completed execution

### PR DESCRIPTION
**Pull request by ajordens**
- `targetReferences` contains complete asg/instance details and is no longer
necessary after an execution has completed

### CODE REVIEW

**Comment by anotherchrisberry** [comment]
One thing to consider: we currently show a very limited set of fields in Deck (name, region, zone) in the `determineTargetReferences` stage details, so we should either blacklist that stage in Deck, or retain just those fields. My preference is the former, as I don't think there's much value in having it in the list of steps on a stage.

**Comment by ajordens** [comment]
Blacklisting seems reasonable to me

**Comment by ajordens** [comment]
@anotherchrisberry removed the `determineTargetReferences` stage as part of https://github.com/spinnaker/deck/pull/2877

File: orca-core/src/main/groovy/com/netflix/spinnaker/orca/config/OrcaConfiguration.groovy [hunk 1]
```diff
@@ -162,15 +165,11 @@ class OrcaConfiguration {
...
-  ExecutionPropagationListener executionPropagationListenerBefore() {
-    // need a dedicated beforeJob listener due to how spring boot ordered listeners
-    new ExecutionPropagationListener(true, false)
-  }
-
-  @Bean
-  ExecutionPropagationListener executionPropagationListenerAfter() {
-    // need a dedicated afterJob listener due to how spring boot ordered listeners
-    new ExecutionPropagationListener(false, true)
+  CompositeExecutionListener executionListeners() {
```
Comment (cfieber) [inline orca-core/src/main/groovy/com/netflix/spinnaker/orca/config/OrcaConfiguration.groovy]: thoughts on just accepting an autowired List<ExecutionListener> 

moving these into explicit construction means that you can't DI a Listener

maybe even down below the places where a CompositeExecutionListener is expected, accept the list of listeners there and internally construct one of these

**Comment by ajordens** [comment]
I consciously did it like it like this because these are the only two
listeners that have an execution order dependency (there are others that
don't).

Rob had done something similar for StageListeners previously.

Personally I think being explicit for now is easier to understand than
@Order.

**Comment by ajordens** [comment]
Don't recall the history behind only having a subset of ExecutionListeners
attached to an Orchestration vs Pipeline.

File: orca-core/src/main/groovy/com/netflix/spinnaker/orca/listeners/ExecutionPropagationListener.groovy [hunk 2]
```diff
@@ -26,20 +26,8 @@ import static com.netflix.spinnaker.orca.ExecutionStatus.*
 @Slf4j
 @CompileStatic
 class ExecutionPropagationListener implements ExecutionListener {
-  private boolean isBeforeJobEnabled = false
-  private boolean isAfterJobEnabled = false
```
Comment (robfletcher) [inline orca-core/src/main/groovy/com/netflix/spinnaker/orca/listeners/ExecutionPropagationListener.groovy]: I like that we're getting rid of this stuff but does it have anything directly to do with this change?

File: orca-core/src/main/groovy/com/netflix/spinnaker/orca/config/OrcaConfiguration.groovy (hunk 1 above)
Comment (robfletcher) [inline orca-core/src/main/groovy/com/netflix/spinnaker/orca/config/OrcaConfiguration.groovy]: You mean you can't add a listener at runtime?

**Pull request merged**